### PR TITLE
Compare tuples without stringifying

### PR DESCRIPTION
**Pull request by bradengroom**
This updates the mysql datastore to compare tuples without stringifying.

### CODE REVIEW

File: pkg/tuple/tuple.go [hunk 1]
```diff
@@ -263,6 +271,11 @@ func Delete(tpl *core.RelationTuple) *core.RelationTupleUpdate {
 	}
 }
 
+func Equal(lhs, rhs *core.RelationTuple) bool {
+	// TODO(jschorr): Use a faster method then string comparison for caveats.
+	return OnrEqual(lhs.ResourceAndRelation, rhs.ResourceAndRelation) && OnrEqual(lhs.Subject, rhs.Subject) && MustStringCaveat(lhs.Caveat) == MustStringCaveat(rhs.Caveat)
```
Comment (bradengroom) [inline pkg/tuple/tuple.go:276]: We still stringify the caveat if everything else matches. There might be some optimizations that could be made here. For example, we could count the keys before stringifying. I'm not sure if this is worth doing.

Comment (vroldanbet) [reply pkg/tuple/tuple.go:276]: There is a function in the codebase that computes a hash of the caveat context. I'm not sure that'd be faster than stringifying tho.

File: pkg/tuple/onr.go [hunk 2]
```diff
@@ -108,3 +108,12 @@ func StringsONRs(onrs []*core.ObjectAndRelation) []string {
 	sort.Strings(onrstrings)
 	return onrstrings
 }
+
+func OnrEqual(lhs, rhs *core.ObjectAndRelation) bool {
+	// Properties are sorted by highest to lowest cardinality to optimize for short-circuiting.
+	return lhs.ObjectId == rhs.ObjectId && lhs.Relation == rhs.Relation && lhs.Namespace == rhs.Namespace
+}
+
+func OnrEqualOrWildcard(tpl, target *core.ObjectAndRelation) bool {
```
Comment (vroldanbet) [inline pkg/tuple/onr.go:117]: do we have tests for this?

**Review by vroldanbet — COMMENTED** [review]
Thanks for your contribution! ✨ Changes look good to me, added some minor feedback

File: pkg/tuple/onr.go (hunk 2 above)
Comment (bradengroom) [reply pkg/tuple/onr.go:117]: Nope! They aren't new really. I just relocated them from `internal/graph/check.go`. I can write tests though

**Comment by bradengroom** [comment]
Added tests for the equality functions.

The caveat hashing function I found looked like it accepts a different type (protobuf struct?) than I have. I wasn’t sure how strongly you felt about using the hashing function, but I’m happy to look into it more if you want.

**Review by vroldanbet — APPROVED** [review]
This looks good to me, thanks for contributing back to the project! 🎉

**Pull request merged**